### PR TITLE
Feature/60 update psha adapter

### LINKED ISSUE #60: update the psha_adapter
psha_adapter implementation needs finishing now that GMM ard configs are done

- [x] #88
- [x] GMCM #56 
- [x] #90
- [x] add a config to the official models
- [x] change `PshaAdapterInterface.__init__()` args to a  `NshmModel` rather than SRM and GMCM logic trees (we're adapting a model to a hazard engine, so this is more logical)
- [x] `OpenquakeSimplePshaAdapter.write_config() ` writes the job.ini based on the config

### PR DESCRIPTION
**Pull request by chrisdicaprio**
closes #60, #88, #56, #90

### CODE REVIEW

File: nzshm_model/hazard_config.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from abc import ABC
```
Comment (chrisdicaprio) [inline nzshm_model/hazard_config.py:1]: move to psha_adapter module

File: nzshm_model/psha_adapter/openquake/simple_nrml.py [hunk 2]
```diff
@@ -216,16 +214,41 @@ def write_config(
...
 
-        target_file = pathlib.Path(destination, 'sources.xml')
-        with open(target_file, 'w') as fout:
-            fout.write(xmlstr)
-        return target_file
+        # job.ini
+        self.hazard_config.set_source_logic_tree_file(sources_file.relative_to(destination))
+        self.hazard_config.set_gsim_logic_tree_file(gmcm_file.relative_to(destination))
+
+        # check that required settings not included in default exist
```
Comment (chrisdicaprio) [inline nzshm_model/psha_adapter/openquake/simple_nrml.py:240]: move to is_complete() method of OQ config class

File: nzshm_model/psha_adapter/psha_adapter_interface.py [hunk 3]
```diff
@@ -26,7 +28,7 @@ def unpack_resources(self, cache_folder: Union[pathlib.Path, str], target_folder
         pass
 
     @abstractmethod
-    def config(self):
+    def sources_document(self):
         """Get the PSHA config file"""
```
Comment (chrisdicaprio) [inline nzshm_model/psha_adapter/psha_adapter_interface.py:32]: update doc string

**Comment by chrisbc** [comment]
this PR was superceded by #101

**Pull request closed without merging**